bin/compBuilds.sh

```diff
@@ -78,6 +78,16 @@ generateBuildConfigs() {
 # -----------------------------------------------------------------------------------------------------------------
 generateBuildConfigs
 
+#setup artifactory/docker pull creds
+if [ ! -z ${USE_PULL_CREDS} ] && [ ${USE_PULL_CREDS} = true ]; then
+  echo "BUILDING PULL CREDS"
+  if [ -z ${CRED_SEARCH_NAME} ]; then
+    buildPromptCreds ${PROJECT_NAMESPACE} ${PULL_CREDS} ${DOCKER_REG} "${CRED_ENVS[@]}" ${DOCKER_USERNAME} ${DOCKER_PASSWORD}
+  else
+    buildCreds ${PROJECT_NAMESPACE} ${CRED_SEARCH_NAME} ${PULL_CREDS} ${DOCKER_REG} "${CRED_ENVS[@]}"
+  fi
+fi
```

Comment:
I am not 100% sure we would want to create/handle pull credentials every time we kick-off processing builds: it is more like a one-off task that is completed before anything else can run. What about extracting this chunk to its own function, so that it could be called independently (`genPullCredentials.sh` maybe ?) and let the user(s) deal with it if they need to?

Reply:
ya that sounds like a good idea. I'll start on it in a bit, just working on email verification

Reply:
moved into initOSProjects.sh